### PR TITLE
sg hack hour 14 oct

### PR DESCRIPTION
**Pull request by jhchabran**
Tees the output of all commands into a `~/.sourcegraph/logs/sg.PID.logs` file that you can use to grep against. 

Perhaps we could have a `sg tail` command that takes that and prints it back so teammate can do things like `sg tail | grep gitserver` and only get the GitServer output as they originally mentionned. 

Perhaps it would be cool to keep the colors somehow. 

## Test plan

Loosely tested locally 🫕

### CODE REVIEW

File: dev/sg/internal/run/command.go [hunk 1]
```diff
@@ -32,6 +32,8 @@ type Command struct {
 	ExternalSecrets map[string]secrets.ExternalSecret `yaml:"external_secrets"`
 	Description     string                            `yaml:"description"`
 
+	w io.Writer
```
Comment (burmudar) [inline dev/sg/internal/run/command.go:35]:
```suggestion
	logWriter io.Writer
```

File: dev/sg/internal/run/pid.go [hunk 2]
```diff
@@ -127,3 +130,35 @@ func writePid() error {
...
+	closer func()
+	path   string
+}
+
+func createLogFile() (*logFile, error) {
+	homeDir, err := os.UserHomeDir()
+	if err != nil {
+		return nil, err
+	}
+
```
Comment (burmudar) [inline dev/sg/internal/run/pid.go:145]:
```suggestion
logDir := filepath.Join(homeDir, ".sourcegraph", "logs")
```
and then use `logDir` for the paths below?

File: dev/sg/internal/run/pid.go [hunk 3]
```diff
@@ -100,7 +102,7 @@ func isPidAlive(pid int32) (bool, error) {
 	return false, err
 }
 
-func writePid() error {
+func writePid(logfile string) error {
```
Comment (burmudar) [inline dev/sg/internal/run/pid.go:105]:
```suggestion
func writePid(logFile string) error {
```

**Comment by mrnugget** [comment]
Closing for now to remove it from my GitHub notifications. We can always reopen.

**Pull request closed without merging**